src/middlewared/middlewared/service_exception.py

```diff
@@ -101,3 +101,8 @@ def adapt_exception(e):
         output = ''.join([stdout, stderr]).rstrip()
 
         return CallError(f'Command {cmd} failed (code {e.returncode}):\n{output}')
+
+
+class MatchNotFound(IndexError):
+    """We be raised when a GET call is issued and there is no matching id"""
```

Comment:
That docstring may be misleanding, that is one of the use cases of that exception, but its generally used in filter list /datastore now.